services/venice-server/src/main/java/com/linkedin/venice/listener/StorageReadRequestHandler.java

```diff
@@ -349,6 +354,21 @@ public void channelRead(ChannelHandlerContext context, Object message) throws Ex
     }
   }
 
+  /**
+   * Best effort check for the purpose of reporting misrouted store version metric when the request errors.
+   */
+  private boolean checkMisroutedStoreVersionRequest(RouterRequest request) {
```

Comment:
Do we need to check partition?

Reply:
Felix and i talked about this in a slack thread and agreed that initially let’s just have a metric that detects misrouted requests due to wrong version. Since checking partition will require a lot more work